url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/gnuplot/GnuPlot.java

```diff
@@ -0,0 +1,47 @@
+package gnuplot;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+
+public class GnuPlot {
+    static String exeFile = "gnuplot/binary/pgnuplot";
+
+    Process p;
+    PrintStream ps;
+    BufferedReader dis;
+
+    public GnuPlot() throws Exception {
+        p = Runtime.getRuntime().exec(exeFile);
+        ps = new PrintStream(p.getOutputStream());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *ps*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).